File: favourite-lyrics-jpa/src/main/java/org/jesperancinha/lyrics/jpa/repository/LyricsRepository.java
package org.jesperancinha.lyrics.jpa.repository;

import org.jesperancinha.lyrics.jpa.model.LyricsEntity;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.UUID;

public interface LyricsRepository extends JpaRepository<LyricsEntity, UUID> {

    void deleteAllByParticipatingArtist(String name);

    LyricsEntity findByParticipatingArtist(String Name);

    LyricsEntity findByLyrics(String Lyrics);
}

File: favourite-lyrics-jpa/src/main/kotlin/org/jesperancinha/lyrics/jpa/repository/LyricsRepository.kt
package org.jesperancinha.lyrics.jpa.repository

import org.jesperancinha.lyrics.jpa.model.LyricsEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface LyricsRepository : JpaRepository<LyricsEntity, UUID?> {
    fun deleteAllByParticipatingArtist(name: String)
    fun findByParticipatingArtist(name: String): LyricsEntity
    fun findByLyrics(lyrics: String): LyricsEntity
}